stan/math/prim/prob/wiener4_lccdf.hpp

```diff
@@ -0,0 +1,454 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+
+#include <stan/math/prim/prob/wiener4_lcdf.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Log of probability of reaching the upper bound in diffusion process
+ *
+ * @tparam T_a type of boundary
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ *
+ * @param a The boundary separation
+ * @param w_value The relative starting point
+ * @param v_value The drift rate
+ * @return log probability to reach the upper bound
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto wiener_prob(const T_a& a, const T_v& v_value,
+                        const T_w& w_value) noexcept {
```

Comment:
I'd remove the `noexcept`s on these. It's unlikely, but for any function that can hold an autodiff type we call `malloc` sometimes and our code to call malloc can throw an exception if the user runs out of memory. The exception is a program terminating one, but with `noexcept` here the program would just crash if it hit the exception instead of reporting the out of memory issue.

Reply:
Ok. For which functions shall I delete the `noexcept`? I now deleted it for `wiener_prob`. When do we use `noexcept`?

Reply:
You only want to use `noexcept` when the function is guranteed to not throw an exception. It just tells the compiler, "I promise I cannot throw an exception here" so then the compiler can do optimizations. 

For anything that uses Eigen matrices we can possibly throw an exception. It's good for simpler functions where you can say for certain there cannot be an exception thrown.

Reply:
Hm. But in this `wiener_prob` function I do not see a term like `Eigen` or `Matrix`. And in the other functions in this file neither. So, every function can keep its `noexcept`?